notebook/services/sessions/handlers.py

```diff
@@ -82,6 +84,11 @@ def post(self):
         location = url_path_join(self.base_url, 'api', 'sessions', model['id'])
         self.set_header('Location', location)
         self.set_status(201)
+        self.log.info(f'model={model}')
+        if 'kernel' in model.keys():
+            if 'name' in model['kernel'].keys():
```

Comment:
can we combine these two if statement?

Reply:
Nope combining these two cannot happen since we need to first check if the 'kernel' key is present in the dict then only check for another key inside the dict referring to the above key.

Reply:
I guess we can write this as 
```
if 'kernel' in model.keys() and 'name' in model['kernel'].keys():
    your code...
```

Reply:
Sourcery is also suggesting the same. You can merge the sourcery refactored PR.

Reply:
Got it, Updated.